cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -132,6 +126,18 @@ public void setAuditPublisher(AuditPublisher auditPublisher) {
    * @throws IOException if there is a problem in making an HTTP request to check if the namespace exists.
    */
   public DatasetMeta get(Id.DatasetInstance instance, List<? extends Id> owners) throws Exception {
+    try {
+      return metaCache.get(instance);
+    } catch (ExecutionException e) {
+      Throwables.propagateIfPossible(e.getCause(), Exception.class);
```

Comment:
Since you have `throws Exception` in the signature, I think you can just do `throw e.getCause();`. BTW, doing `e.getCause()`, however, will lose the information about the `metaCache.get` call in the stacktrace.